djangocms_versioning/models.py

```diff
@@ -99,12 +99,19 @@ def delete(self, using=None, keep_parents=False):
         """Deleting a version deletes the grouper
         as well if we are deleting the last version."""
 
+        def get_grouper_name(ContentModel, grouper):
+            GrouperModel = grouper._meta.model
+            for field in ContentModel._meta.fields:
+                if getattr(field, 'related_model', None) == GrouperModel:
+                    return field.name
+
+        grouper = self.grouper
         ContentModel = self.content._meta.model
-        grouper_name = self.grouper._meta.model_name
+
+        grouper_name = get_grouper_name(ContentModel, grouper)
```

Comment:
minor thing, why create grouper when it's just a reference to self.grouper? Compounded by having self.grouper.pk on the line below too.

Reply:
You asked exactly the same thing the last time. :)

grouper is not a reference. It is its own object and we need it once we've
done the deletion.